file/file-alfresco/src/main/java/uk/gov/justice/services/file/alfresco/sender/AlfrescoFileSender.java

```diff
@@ -29,21 +32,23 @@
     String alfrescoUploadUser;
 
     @Inject
-    ObjectMapper objectMapper;
-
-    @Inject
     AlfrescoRestClient restClient;
 
-
     @Override
     public FileData send(final String fileName, final byte[] content) {
-        final MultivaluedHashMap<String, Object> headers = new MultivaluedHashMap<>();
-        headers.put(ID, Collections.singletonList(alfrescoUploadUser));
-
-        restClient.post(alfrescoUploadPath, MULTIPART_FORM_DATA_TYPE, headers, entity(content, MULTIPART_FORM_DATA_TYPE));
-
-        //TODO: build response
-
-        return null;
+        try {
+            final Response response = restClient.post(alfrescoUploadPath, MULTIPART_FORM_DATA_TYPE, headersWithUserId(alfrescoUploadUser), entity(content, MULTIPART_FORM_DATA_TYPE));
+            final String responseEntity = response.readEntity(String.class);
+            if (response.getStatusInfo() != OK || isEmpty(responseEntity)) {
+                //Alfresco is *very* accepting - failed exceptions represent service outages/problems only.
+                throw new FileOperationException(String.format("Error while uploading document. Code:%d, Reason:%s", response.getStatus(), response.getStatusInfo().getReasonPhrase()));
+            }
+            final Object responseDocument = defaultConfiguration().jsonProvider().parse(responseEntity);
+            final String alfrescoAssetId = JsonPath.read(responseDocument, "$.nodeRef").toString().replace("workspace://SpacesStore/", "");
+            final String mimeType = JsonPath.read(responseDocument, "$.fileMimeType");
```

Comment:
using constants for json paths would obfuscate the code a little, I usually prefer to have them directly in code, if they are not reused elsewhere.

extracting private method done